keps/0009-operator-toolkit.md

```diff
@@ -106,93 +100,80 @@ An operator package is a folder that contains all of the manifests needed to cre
 
 #### operator.yaml
 
-`operator.yaml` is the base definition of an operator. It follows the following format, extracted from the MySQL example:
+`operator.yaml` is the base definition of an operator.  It follows the following reference format.
 
 ```yaml
-name: operator
-description: operator
-version: "5.7"
-kudoVersion: ">= 0.2.0"
-kubeVersion: ">= 1.14"
+name: str(max=253, lowercase alphanumeric characters, -, and .)
+description: str(required=False)
+version: semver(desc='version of operator')
+appVersion: str(desc='component controlled version', required=False)
+kudoVersion: semver(desc='minVersion of KUDO')
+kubernetesVersion: semver(desc='minVersion of Kubernetes')
+maintainers: array(required=False, object=Maintainer)
+  - name: str()
+    email: email()
+  - name: str()
+    email: email()
+url: url(required=False)
+tasks: map[string]TaskSpec
+  str(max 253, desc='name of task'):
+    resources: array(str)
+      - str(desc='name of resource')
+plans: map[string]Plan
+  str(max 253, desc='name of plan'):
+    strategy: ordering(*serial|parallel, desc='determines how to order phases', required=False)
```

Comment:
should we also document what is the default somewhere since it's not required?

Reply:
Currently this is required in code... this was a suggestion to make it not required and providing default information is super useful... that was the intent of `*` in `*serial` but that could be better.